### DISCUSSION TITLE
Error when trying to sort by column

### DISCUSSION THREAD

**Discussion by gintas** [dbeaver/dbeaver#325] [Q&A] [answered]
It looks like the DBeaver parser can't deal with WITH clauses for the SQL server dialect. I tried to sort by a column (by clicking on the column header) for this query:

SELECT TOP 5000 \* FROM SomeBigTable t WITH (NOLOCK)

And got this:

DBCException: Can't apply query filter
  DBException: SQL parse error
    JSQLParserException
      net.sf.jsqlparser.parser.ParseException: ParseException: Encountered " "WITH" "WITH "" at line 2, column 40.
Was expecting one of:
    <EOF> 
    "JOIN" ...
    "LEFT" ...
    "CROSS" ...
    "FULL" ...
    "WHERE" ...
    "FOR" ...
    "GROUP" ...
    "INNER" ...
    "RIGHT" ...
    "HAVING" ...
    "NATURAL" ...
    "START" ...
    "CONNECT" ...
    ";" ...
    "," ...
    "LEFT" ...
    "RIGHT" ...
    "FULL" ...
    "INNER" ...
    "NATURAL" ...
    "CROSS" ...
    "JOIN" ...
    "," ...
    "WHERE" ...
    "START" ...
    "CONNECT" ...
    "GROUP" ...
    "HAVING" ...
    "FOR" ...

**Comment by serge-rider** [comment, marked as answer]
Yes, DBeaver doesn't support it. In fact it doesn't support pretty many elements in different SQL dialects. We don't have plans to improve semantic SQL parsing in near future.

As a workaround you can disable server-side ordering in preferences.
